core/dbt/task/test.py

```diff
@@ -387,14 +397,24 @@ class TestTask(RunTask):
     def raise_on_first_error(self) -> bool:
         return False
 
+    @property
+    def resource_types(self) -> List[NodeType]:
+        resource_types: Collection[NodeType] = resource_types_from_args(
+            self.args, set(TEST_NODE_TYPES), set(TEST_NODE_TYPES)
+        )
+
+        # filter out any non-test node types
+        resource_types = [rt for rt in resource_types if rt in TEST_NODE_TYPES]
+        return list(resource_types)
+
```

Comment:
This is basically the same thing as [here](https://github.com/dbt-labs/dbt-core/blob/3e437a67344469d51d7f2d4a4d0338a45c1de261/core/dbt/task/list.py#L179-L188) and [here](https://github.com/dbt-labs/dbt-core/blob/3e437a67344469d51d7f2d4a4d0338a45c1de261/core/dbt/task/build.py#L85-L94).